File: src/convex/geometry/Point.java
package convex.geometry;

import java.util.Scanner;

public class Point {
    private double x, y;
    public Point() throws Exception {
        Scanner in = new Scanner(System.in);
        System.out.print("x -> ");
        x = Double.valueOf(in.nextLine());
        System.out.print("y -> ");
        y = Double.valueOf(in.nextLine());
    }
    public double getX() {
        return x;
    }
    public double getY() {
        return y;
    }
    public boolean light(Point a, Point b) {
        double s = Basics.area(a, b, this);
        return s < 0.0 || (s == 0 && ! inside(a, b));
    }
    public boolean inside(Point a, Point b) {
        return (a.x <= x && x <= b.x || a.x >= x && x >= b.x) && (a.y <= y && y <= b.y || a.y >= y && y >= b.y);
    }
}

File: Convex/Geometry/Point.cs
using System;

namespace Convex.Geometry
{
    public class Point
    {
        public double X { get; }
        public double Y { get; }

        public Point()
        {
            Console.Write("Enter X: ");
            X = Convert.ToDouble(Console.ReadLine());
            Console.Write("Enter Y: ");
            Y = Convert.ToDouble(Console.ReadLine());
        }

        public bool IsLighted(Point a, Point b)
        {
            var s = Basics.Area(a, b, this);
            return s < 0.0 || (s == 0 && !IsInside(a, b));
        }

        public bool IsInside(Point a, Point b)
        {
            return (a.X <= X && X <= b.X || a.X >= X && X >= b.X) && (a.Y <= Y && Y <= b.Y || a.Y >= Y && Y >= b.Y);
        }
    }
}
